service-base/src/main/java/org/eclipse/hono/service/limiting/DefaultConnectionLimitManager.java

```diff
@@ -61,6 +61,10 @@ private int autoconfigureConnectionLimit() {
 
         final int recommendedLimit = strategy.getRecommendedLimit();
 
+        if (recommendedLimit == 0) {
+            throw new RuntimeException("Not enough memory to handle connections. To override this check, configure a connection limit.");
```

Comment:
Please throw a more specific exception and also document it in the constructor's JavaDoc.

Reply:
Like this?

Reply:
Sorry, I didn't make myself clear enough. I would have been completely content with e.g. throwing an `IllegalStateException` here. However, using a newly designed exception works as well, of course :-)